src/model/include/iDynTree/ModelTestUtils.h

```diff
@@ -8,18 +8,34 @@
 #include <iDynTree/Model.h>
 #include <iDynTree/FixedJoint.h>
 #include <iDynTree/RevoluteJoint.h>
+#include <iDynTree/RevoluteSO2Joint.h>
 #include <iDynTree/PrismaticJoint.h>
 #include <iDynTree/FreeFloatingState.h>
 #include <iDynTree/LinkState.h>
 
 #include <iDynTree/TestUtils.h>
 
 #include <cassert>
+#include <cmath>
 #include "IJoint.h"
 
 namespace iDynTree
 {
 
+/**
+ * Bitfield enum for selecting joint types in random model generation.
+ */
+enum JointTypes : unsigned int
+{
+    JOINT_FIXED = 1 << 0,        // 0001 - FixedJoint
+    JOINT_REVOLUTE = 1 << 1,     // 0010 - RevoluteJoint
+    JOINT_PRISMATIC = 1 << 2,    // 0100 - PrismaticJoint
+    JOINT_REVOLUTE_SO2 = 1 << 3  // 1000 - RevoluteSO2Joint
+};
+
+// Default joint types: Fixed, Revolute, and Prismatic
+const unsigned int DEFAULT_JOINT_TYPES = JOINT_FIXED | JOINT_REVOLUTE | JOINT_PRISMATIC;
```

Comment:
```suggestion
// Simple joint types: Fixed, Revolute, and Prismatic
const unsigned int SIMPLE_JOINT_TYPES = JOINT_FIXED | JOINT_REVOLUTE | JOINT_PRISMATIC;
```

I am not a big fan of `DEFAULT_JOINT_TYPES`, as it is not clear what `default` refers to. Perhaps we can just use `SIMPLE_JOINT_TYPES`, as I think that everyone can agree that fixed, revolute and prismatic are simple, while REVOLUTE_SO2 or Spherical are not simple.

Reply:
Can we also define `const unsigned int ALL_JOINT_TYPES` with all the joint types?